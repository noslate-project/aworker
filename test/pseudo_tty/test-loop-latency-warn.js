// META: flags=--long-task-threshold-ms=15
'use strict';

function test() {
  const start = Date.now();
  while (Date.now() - start <= 35) {
    /** busy loop */
  }
}

function main() {
  test();
  // Schedule two task in one loop tick.
  setTimeout(function timer1() {
    test();
  }, 1000);
  setTimeout(function timer2() {
    test();
  }, 1000);
}

main();
