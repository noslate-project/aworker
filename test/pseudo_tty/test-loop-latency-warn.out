Trace: Loop latency reached limit
  at test (*/test/pseudo_tty/test-loop-latency-warn.js:6:3)
  at main (*/test/pseudo_tty/test-loop-latency-warn.js:12:3)
  at * (*/test/pseudo_tty/test-loop-latency-warn.js:18:1)
  *
Trace: Loop latency reached limit
  at test (*/test/pseudo_tty/test-loop-latency-warn.js:6:3)
  at (null) (*/test/pseudo_tty/test-loop-latency-warn.js:14:5)
  *
